dask/dataframe/groupby.py

```diff
@@ -610,12 +627,87 @@ def _aca_agg(self, token, func, aggfunc=None, split_every=None, split_out=1):
+        cumpart_raw = map_partitions(_apply_chunk, self.obj, *index,
+                                     chunk=chunk,
+                                     columns=columns,
+                                     token=name1,
+                                     meta=meta)
+        cumpart_ext = (cumpart_raw.to_frame()
+                       if isinstance(meta, pd.Series)
+                       else cumpart_raw).assign(**{i: self.obj[i]
+                                                   for i in index})
+
+        name2 = '{0}{1}-take-last'.format(self._token_prefix, token)
+        cumlast = map_partitions(_apply_chunk, cumpart_ext, *index,
+                                 columns=0 if columns is None else columns,
+                                 chunk=M.last,
+                                 meta=meta,
+                                 token=name2)
+
+        name = '{0}{1}'.format(self._token_prefix, token)
+        cname = '{0}{1}-cum-last'.format(self._token_prefix, token)
+
+        # aggregate cumulated partisions and its previous last element
+        dask = {}
+        dask[(name, 0)] = (cumpart_raw._name, 0)
+
+        for i in range(1, self.obj.npartitions):
+            # store each cumulative step to graph to reduce computation
+            if i == 1:
+                dask[(cname, i)] = (cumlast._name, i - 1)
+            else:
+                # aggregate with previous cumulation results
+                dask[(cname, i)] = (_cum_agg_filled,
+                                    (cname, i - 1), (cumlast._name, i - 1),
+                                    aggregate, neutral)
+            dask[(name, i)] = (_cum_agg_aligned,
+                               (cumpart_ext._name, i), (cname, i),
+                               index, 0 if columns is None else columns,
+                               aggregate, neutral)
+        return new_dd_object(merge(dask, cumpart_ext.dask, cumlast.dask),
+                             name, chunk(self._meta), self.obj.divisions)
+
+    @derived_from(pd.core.groupby.GroupBy)
+    def cumsum(self, axis=0):
+        if axis:
+            return self.obj.cumsum(axis=axis)
+        else:
+            return self._cum_agg('cumsum',
+                                 chunk=M.cumsum,
+                                 aggregate=M.add,
+                                 neutral=0)
+
+    @derived_from(pd.core.groupby.GroupBy)
+    def cumprod(self, axis=0):
+        if axis:
+            return self.obj.cumprod(axis=axis)
+        else:
+            return self._cum_agg('cumprod',
+                                 chunk=M.cumprod,
+                                 aggregate=M.mul,
+                                 neutral=1)
+
+    @derived_from(pd.core.groupby.GroupBy)
+    def cumcount(self, axis=None):
+        return self._cum_agg('cumcount',
+                             chunk=M.cumcount,
+                             aggregate=_add_plus_one,
```

Comment:
Slight request to give a more semantic name. Perhaps `cumcount_aggregate`, or something.